app/forms/validators.py

```diff
@@ -222,17 +222,15 @@ def __init__(self, message=None):
         self.message = message or error_messages["INVALID_DATE"]
 
     def __call__(self, form, field):
-
-        if not form.data or not re.match(r"\d{4}$", str(form.year.data)):
+        if not form.data or not hasattr(form, "year"):
```

Comment:
The `year` attribute doesn't need to be checked here, as it will always be present in a date form. 
```suggestion
        if not form.data:
```

Reply:
Removed, thanks!